bundle-hack/update_csv.py

```diff
@@ -0,0 +1,172 @@
+        if filename.endswith('clusterserviceversion.yaml'):
+            return os.path.join(d, filename)
+
+def add_required_annotations(m: dict) -> None:
+    """
+    Adds required annotations to the CSV file content represented as a dictionary.
+    Errors out if 'metadata' or 'annotations' does not exist.
+
+    :param m: A dictionary representing the operator CSV manifest.
+    """
+    required_annotations = {
+        "features.operators.openshift.io/disconnected": "true",
+        "features.operators.openshift.io/fips-compliant": "true",
+        "features.operators.openshift.io/proxy-aware": "true",
+        "features.operators.openshift.io/tls-profiles": "false",
+        "features.operators.openshift.io/token-auth-aws": "false",
+        "features.operators.openshift.io/token-auth-azure": "false",
+        "features.operators.openshift.io/token-auth-gcp": "false"
+    }
+
+    if 'metadata' not in m:
+        sys.exit("Error: 'metadata' does not exist in the CSV content.")
+
+    if 'annotations' not in m['metadata']:
+        sys.exit("Error: 'annotations' does not exist within 'metadata' in the CSV content.")
+
+    for key, value in required_annotations.items():
+        m['metadata']['annotations'][key] = value
+
+    print("Successfully added required annotations.")
+
+def replace_version(m: dict) -> None:
+    manifest_version = m['spec']['version']
+    print(f"Updating version references from {manifest_version}",
+          f"to {args.version} in manifest.")
+    m['metadata']['name'] = m['metadata']['name'].replace(manifest_version, args.version)
+    m['metadata']['annotations']['olm.skipRange'] = m['metadata']['annotations']['olm.skipRange'].replace(manifest_version, args.version)
+    m['spec']['replaces'] = 'compliance-operator.v' + manifest_version
+    m['spec']['version'] = args.version
+    print(f"Successfully updated the operator version references from",
+          f"{manifest_version} to {args.version} in manifest.")
+
+
+def replace_icon(m: dict) -> None:
+    """Replace the upstream icon with a Red Hat branded image.
+
+    Perform an in-place update of the icon data on the manifest.
+
+    manifest(dict): A dictionary representing the operator CSV manifest
+    """
+    icon_path = "icons/icon.png"
+    with io.open(icon_path, "rb") as f:
+        base64_encoded_icon = base64.b64encode(f.read())
+    icons = [{"base64data": base64_encoded_icon.decode(), "mediatype": "image/png"}]
+    m['spec']['icon'] = icons
+    print(f"Successfully updated the operator image to use icon in {icon_path}.")
+
+
+def replace_images(m: dict) -> None:
+
+    CO_OPERATOR_IMAGE_PULLSPEC = "quay.io/redhat-user-workloads/ocp-isc-tenant/operator"
+    CO_CONTENT_IMAGE_PULLSPEC = "quay.io/redhat-user-workloads/ocp-isc-tenant/content"
+    CO_OPENSCAP_IMAGE_PULLSPEC = "quay.io/redhat-user-workloads/ocp-isc-tenant/scanner"
+    CO_MUST_GATHER_IMAGE_PULLSPEC = "quay.io/redhat-user-workloads/ocp-isc-tenant/must-gather"
```

Comment:
@Vincent056 You can refer to actual builds done in Konflux.

Also, these components should be renamed to include the application name.